src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -488,11 +488,12 @@ protected void writeToNBT(@NotNull final NBTTagCompound compound)
 
         //  Achievements
         @NotNull final NBTTagList achievementsTagList = new NBTTagList();
-        for (@NotNull final Achievement achievement : this.colonyAchievements)
+        for (@NotNull final Advancement achievement : this.colonyAchievements)
         {
             @NotNull final NBTTagCompound achievementCompound = new NBTTagCompound();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "achievementCompound". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "achievementCompound" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1481)